packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -160,6 +160,28 @@ export class CallContext {
     });
   }
 
+  /* @conditional-compile-remove(PSTN-calls) */
+  public setCallHold(callId: string): void {
+    const latestCallId = this._callIdHistory.latestCallId(callId);
+    this.modifyState((draft: CallClientState) => {
+      const call = draft.calls[latestCallId];
+      if (call && call.state !== 'RemoteHold') {
+        this.setCallState(latestCallId, 'LocalHold');
+      }
+    });
+  }
+
+  /* @conditional-compile-remove(PSTN-calls) */
+  public setCallResume(callId: string): void {
```

Comment:
Where will these functions be consumed?

Reply:
does it seem to assume when we resume the call we are on "Connected"? Is it possible to somehow read the last known state of the call instead and to use that value?